### PR TITLE
20220912_BugFixed_/learn/foundations/from-react-to-nextjs

### PR DESCRIPTION
**Pull request by Nicolas19921107**
20220910_嘗試 pull draft request

### CODE REVIEW

File: pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx [hunk 1]
```diff
@@ -1,12 +1,222 @@
...
+import Blockquote from "@components/Blockquote";
+import Button from "@components/Button";
+
+# 從 React 到 Next.js
+
+##　開始您的 Next.js
+
+在安裝 Next.js 到您的專案前，您不再需要透過手動加入 `react` 與 `react-dom` 的 script 標籤的方式引入 React
+
+取而代之的是，您可以在本地端使用套件管理工具 `npm` 來安裝這些套件
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx:10]:
```suggestion
在安裝 Next.js 到您的專案前，您不再需要透過手動加入 `react` 與 `react-dom` 的 script 標籤的方式引入 React。取而代之的是，您可以在本地端使用套件管理工具 `npm` 來安裝這些套件。
```

File: pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx [hunk 2]
```diff
@@ -1,12 +1,222 @@
...
+  [最低版本需求](/docs/upgrading#minimum-nodejs-version)，[您可以在此進行下載](https://nodejs.org/en/)
+</Blockquote>
+
+在準備開發前，請先新增一個內容為空物件 `{}` 的 `package.json` 文檔。
+
+```
+    //package.json
+    {
+
+    }
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx:23]:
```suggestion
//package.json
{

}
```

File: pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx [hunk 3]
```diff
@@ -1,12 +1,222 @@
...
+          <li key={name}>{name}</li>
+        ))}
+      </ul>
+
+      <button onClick={handleClick}>Like ({likes})</button>
+    </div>
+  );
+}
+```
+此時在 `index.html` 中的程式碼均為 JSX 語法，為此您可將 `index.html` 的檔案類型由 .html 替換為 `.js` 或 `.jsx`。
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx:83]: 在 Markdown 或 MDX 語法中，不同類型的區塊（比如說 Code Block 和一般文字）中間要空一行。

File: pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx [hunk 4]
```diff
@@ -1,12 +1,222 @@
...
+
+      <button onClick={handleClick}>Like ({likes})</button>
+    </div>
+  );
+}
+```
+
+從表面上來看，這些過程只是減少了幾行程式碼，但它確實幫助凸顯了一些事情：React 是一種**函式庫**，它提供了構築現代互動式 UI 的**一切基本**事物，但仍需要透過一些工序將您製作的 UI 整合至應用程式中。
+
+看完前面對於程式碼上的變更，您或許已經理解使用 Next.js 的好處。您移除了 Babel Script， 體驗了不再需要多餘思考而配置的複雜工具。您也觀察到了[快速刷新]( Fast Refresh )的實際效果，而這些只是 Next.js 在眾多開發者體驗特色中的其中一項。
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx:198]:
```suggestion
看完前面對於程式碼上的變更，您或許已經理解使用 Next.js 的好處。您移除了 Babel Script， 體驗了不再需要多餘思考而配置的複雜工具。您也觀察到了快速重新整理（ Fast Refresh ）的實際效果，而這些只是 Next.js 在眾多開發者體驗特色中的其中一項。
```

File: pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx [hunk 5]
```diff
@@ -1,12 +1,222 @@
...
+
+
+<Button
+  content="返回上一頁 ←"
+  url="/learn/foundations/about-nextjs/index"
+  basicStyle="text-white bg-[#0070f3]"
+  hoverStyle="hover:opacity-90"
+/>
+<Button
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx:207]:
```suggestion
/>

<Button
```

File: pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx [hunk 6]
```diff
@@ -1,12 +1,222 @@
...
 <Pagination
   hrefs={[
-    '/learn/foundations/from-react-to-nextjs',
-    '/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs',
-    '/learn/foundations/from-react-to-nextjs/next-steps'
+    "/learn/foundations/from-react-to-nextjs",
+    "/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs",
+    "/learn/foundations/from-react-to-nextjs/next-steps",
   ]}
 />
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx:222]: 這個應該放在大標題下面

**Review by yuaanlin — CHANGES_REQUESTED** [review]
Hi, thanks for the contribution! There is some changes needed to let this PR be merged. Please see the comments below.

**Review by Nicolas19921107 — COMMENTED** [review]
已調整

File: pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx [hunk 7]
```diff
@@ -1,12 +1,216 @@
...
+# 從 React 到 Next.js
 
 <Pagination
   hrefs={[
-    '/learn/foundations/from-react-to-nextjs',
-    '/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs',
-    '/learn/foundations/from-react-to-nextjs/next-steps'
+    "/learn/foundations/from-react-to-nextjs",
+    "/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs",
+    "/learn/foundations/from-react-to-nextjs/next-steps",
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx:10]:
```suggestion
    '/learn/foundations/from-react-to-nextjs',
    '/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs',
    '/learn/foundations/from-react-to-nextjs/next-steps',
```

File: pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx [hunk 8]
```diff
@@ -1,12 +1,216 @@
...
   ]}
 />
 
-<NotTranslateYet/>
+## 開始你的 Next.js
+
+在安裝 Next.js 到你的專案前，你不再需要透過手動加入 `react` 與 `react-dom` 的 script 標籤的方式引入 React
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx:16]:
```suggestion
在安裝 Next.js 到你的專案前，你不再需要透過手動加入 `react` 與 `react-dom` 的 script 標籤的方式引入 React 。
```

File: pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx [hunk 9]
```diff
@@ -1,12 +1,216 @@
...
+取而代之的是，你可以在本地端使用套件管理工具 `npm` 來安裝這些套件
+
+<Blockquote>
+  小提醒：你需要在電腦中安裝 Node.js
+  [最低版本需求](/docs/upgrading#minimum-nodejs-version)，[可以在此進行下載](https://nodejs.org/en/)
+</Blockquote>
+
+在準備開發前，請先新增一個內容為空物件 `{}` 的 `package.json` 文檔。
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx:24]:
```suggestion
在準備開發前，請先新增一個內容為空物件 `{}` 的 `package.json` 檔案。
```

File: pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx [hunk 10]
```diff
@@ -1,12 +1,216 @@
...
+
+```html
+//package.json
+{
+
+}
+```
+
+接著，請在你的終端機中運行 `npm install react react-dom next`。
+當安裝程序完成後，你可以在 `package.json` 文檔中，查看專案所有的相依套件與應用程式資訊：
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx:34]:
```suggestion
當安裝程序完成後，你可以在 `package.json` 檔案中，查看專案所有的相依套件資訊：
```

File: pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx [hunk 11]
```diff
@@ -1,12 +1,216 @@
...
+     //   "next": "^11.1.0",
+     //   "react": "^17.0.2",
+     //   "react-dom": "^17.0.2"
+     // }
+   }
+```
+
+## 運行開發環境
+
+你可透過終端機運行 `npm run dev` 並開啟瀏覽器前往 [localhost:3000](localhost:3000) 以查看應用程式是否正常運作。接下來，你可以針對程式碼試著做些調整並儲存
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx:119]:
```suggestion
你可透過終端機運行 `npm run dev` 並開啟瀏覽器前往 [localhost:3000](localhost:3000) 以查看應用程式是否正常運作。接下來，你可以針對程式碼試著做些調整並儲存。
```

File: pages/learn/foundations/from-react-to-nextjs/index.mdx [hunk 12]
```diff
@@ -1,11 +1,70 @@
...
+# 從 React 到 Next.js
 
 <Pagination
   hrefs={[
-    '/learn/foundations/from-react-to-nextjs',
-    '/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs',
-    '/learn/foundations/from-react-to-nextjs/next-steps'
+    "/learn/foundations/from-react-to-nextjs",
+    "/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs",
+    "/learn/foundations/from-react-to-nextjs/next-steps"
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/index.mdx:9]:
```suggestion
    '/learn/foundations/from-react-to-nextjs',
    '/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs',
    '/learn/foundations/from-react-to-nextjs/next-steps'
```

File: pages/learn/foundations/from-react-to-nextjs/next-steps.mdx [hunk 13]
```diff
@@ -1,11 +1,41 @@
...
+
+總而言之，你已探索了 React 與 Next.js 的基礎知識，並從簡單的 React 應用程式轉移到了 Next.js 的應用程式
+現在你可以在 Next.js 的旅程中選擇下一步。你可以：
+
+1. 建立你的第一支應用程式以學習如何運用 Next.js - 這個課程將會介紹 Next.js 的主要功能，並透過建立更為複雜的專案以提供你進行練習
+2. 繼續本課程，更深入地理解 Next.js 的運作原理
+
+在接下來的課程中，我們將會透過相關的網頁開發概念去探索 Next.js 的運作原理。
+
+熟悉這些觀念有助你擴展相關的知識基礎，並可以更容易地學習到更多關於 Next.js 的進階功能
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/next-steps.mdx:25]:
```suggestion
熟悉這些觀念有助你擴展相關的知識基礎，並可以更容易地學習到更多關於 Next.js 的進階功能。
```

File: pages/learn/foundations/from-react-to-nextjs/next-steps.mdx [hunk 14]
```diff
@@ -1,11 +1,41 @@
...
+/>
+
+<Button
+  content="前往下一個課程 →"
+  url="/learn/foundations/how-nextjs-works/index"
+  basicStyle="text-white bg-[#0070f3]"
+  hoverStyle="hover:opacity-90"
+/>
+
+<TranslatedBy githubUserNames={["st880517"]} />
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/next-steps.mdx:41]:
```suggestion
<TranslatedBy githubUserNames={["Nicolas19921107"]} />
```

File: pages/learn/foundations/from-react-to-nextjs/index.mdx [hunk 15]
```diff
@@ -1,11 +1,70 @@
...
+現在，讓我們瞧瞧如何將一個簡單的 React 應用程式轉移到 Next.js 中。
+
+<Button
+  content="前往課程 →"
+  url="/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs"
+  basicStyle="text-white bg-[#0070f3]"
+  hoverStyle="hover:opacity-90"
+/>
+
+<TranslatedBy githubUserNames={["st880517"]} />
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/index.mdx:70]:
```suggestion
<TranslatedBy githubUserNames={["Nicolas19921107"]} />
```

File: pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx [hunk 16]
```diff
@@ -1,12 +1,216 @@
...
+/>
+
+<Button
+  content="前往下一頁 →"
+  url="/learn/foundations/from-react-to-nextjs/next-steps"
+  basicStyle="text-white bg-[#0070f3]"
+  hoverStyle="hover:opacity-90"
+/>
 
+<TranslatedBy githubUserNames={["st880517"]} />
```
Comment (yuaanlin) [inline pages/learn/foundations/from-react-to-nextjs/getting-started-with-nextjs.mdx:216]:
```suggestion
<TranslatedBy githubUserNames={["Nicolas19921107"]} />
```

**Review by yuaanlin — CHANGES_REQUESTED** [review]

**Review by yuaanlin — APPROVED** [review]

**Pull request merged**